python/cudf/dataframe/string.py

```diff
@@ -108,9 +108,43 @@ def cat(self, others=None, sep=None, na_rep=None):
             (same type as caller) of str dtype is returned.
         """
         from cudf.dataframe import Series, Index
+        def is_list_like(others):
+            if isinstance(others, (Series, list, tuple, Index)) and not isinstance(others, (str, bytes)):
+                return True
+            else:
+                return False
+
         if isinstance(others, (Series, Index)):
+            # If others is just another Series/Index, great go ahead with concatenation
             assert others.dtype == np.dtype('object')
             others = others.data
+        elif is_list_like(others):
+            # If others is a list-like object (in our case lists & tuples)just another Series/Index, great go ahead with concatenation
+            if all(is_list_like(x) for x in others):
+                # Internal elements in others list should also be list-like and not a regular string/byte
+                others = list(others)
+                first = others.pop(0)
```

Comment:
Is `.pop(...)` not a standard function for list-likes? Do we need to convert to a list above?

Reply:
`others` can be either a tuple or list, Since tuples don't have `.pop`, I had to convert it to list.

Reply:
Talked offline about this one, going to refactor this segment to not need to pop and to not use the `all(...)` call above and to instead check the first element only and then choose an appropriate logically branch as needed depending on if it's a string or an array-like.

Reply:
Made the changes as we discussed and testing it @kkraus14